quixstreams/sinks/community/tdengine/sink.py

```diff
@@ -328,29 +349,72 @@ def write(self, batch: SinkBatch):
                         k: float(v) if isinstance(v, int) else v
                         for k, v in fields.items()
                     }
-                ts = (
-                    value[time_key]
-                    if time_key is not None and time_key in value
-                    else item.timestamp
-                )
+                # check if fields is empty
+                if not fields:
+                    raise ValueError(
+                        f'No fields found in the record for supertable "{_measurement}" '
+                        f"and subtable name '{_subtable_name}', "
+                        f"record value: {copied_value}"
+                    )
+                
+                tags["__subtable"] = _subtable_name
```

Comment:
Can subtable_name be None/null/"" ?

Reply:
> Can subtable_name be None/null/"" ?

If subtable_name is None or an empty string, it will be ignored when generating the line protocol.